File: src/main/java/com/neomind/supplierchallenge/service/SupplierService.java
package com.neomind.supplierchallenge.service;

import java.util.Collection;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.neomind.supplierchallenge.entity.Supplier;
import com.neomind.supplierchallenge.exception.SupplierNotFoundException;
import com.neomind.supplierchallenge.repository.SupplierRepository;

@Service
public class SupplierService {
	
	@Autowired
	SupplierRepository repository;
	
	public Collection<Supplier> findAll() {
		return repository.findAll();
	}
	
	public Supplier findById(Long id) throws SupplierNotFoundException {
		return repository.findById(id).orElseThrow(SupplierNotFoundException::new);
	}
	
	public Supplier save(Supplier supplier) {
		return repository.save(supplier);
	}

	public Supplier update(Long id, Supplier supplier) throws SupplierNotFoundException {
		Supplier existingSupplier = findById(id);
		
		existingSupplier.setName(supplier.getName());
		existingSupplier.setComment(supplier.getComment());
		existingSupplier.setEmail(supplier.getEmail());
		existingSupplier.setCnpj(supplier.getCnpj());
		
		return save(existingSupplier);
	}

	public void delete(Long id) throws SupplierNotFoundException {
		repository.delete(findById(id));
	}

}

File: src/main/resources/static/service/supplier.service.js
angular
    .module("supplier-component")
    .service('SupplierService', ['$http', function ($http) {

        this.findAll = () => {
            return $http({
                method: 'GET',
                url: 'suppliers'
            });
        };

        this.insert = (supplier) => {
            return $http({
                method: 'POST',
                url: 'suppliers',
                data: supplier
            });
        };

        this.findById = (id) => {
            return $http({
                method: 'GET',
                url: 'suppliers/' + id
            });
        };
        
        this.update = (id, supplier) => {
            return $http({
                method: 'PUT',
                url: 'suppliers/' + id,
                data: supplier
            })
        };

        this.delete = (id) => {
            return $http({
                method: 'DELETE',
                url: 'suppliers/' + id
            })
        };

    }]);